Clearer error messages needed on validation error

In its current state, when the `validate` goal is activated in an execution, formatting errors create the following message:

```
[ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.0.0:validate (default) on project nd4j-api: File '/home/raver119/develop/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java' format doesn't match! -> [Help 1]
```

Without more insights, it's hard to figure out what the formatter does not like — it would be useful to have a more specific error message, with a line number pointing to the first error.

I do agree but at the root it's just saying formatting was not performed by developer and then committed to repo.  So solution is for developer to run build and commit changes.

Will keep this open though as it has merit.

Get Outlook for Android<https://aka.ms/ghei36>


________________________________
From: François Garillot <notifications@github.com>
Sent: Monday, March 6, 2017 5:19:35 PM
To: revelc/formatter-maven-plugin
Cc: Subscribed
Subject: [revelc/formatter-maven-plugin] Clearer error messages (#172)


In its current state, when the validate goal is activated in an execution, formatting errors create the following message:

[ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.0.0:validate (default) on project nd4j-api: File '/home/raver119/develop/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java' format doesn't match! -> [Help 1]
```
Without insights, it's hard to figure out what the formatter does not like — it would be useful to have a more specific error message, with a line number pointing to the first error.


—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/revelc/formatter-maven-plugin/issues/172>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AA7hoyffeyoEcfbOnDqxaOqE_dFFBRYqks5rjIZ3gaJpZM4MUxft>.

--- a/maven-plugin/src/main/java/net/revelc/code/formatter/ValidateMojo.java
+++ b/maven-plugin/src/main/java/net/revelc/code/formatter/ValidateMojo.java
@@ -1,5 +1,5 @@
 /**
- * Copyright 2010-2016. All work is copyrighted to their respective
+ * Copyright 2010-2017. All work is copyrighted to their respective
  * author(s), unless otherwise stated.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
@@ -59,7 +59,8 @@ public class ValidateMojo extends FormatterMojo {
         super.doFormatFile(file, rc, hashCache, basedirPath, true);
 
         if (rc.successCount != 0) {
-            throw new MojoFailureException("File '" + file + "' format doesn't match!");
+            throw new MojoFailureException("File '" + file
+                    + "' has not been previously formatted.  Please format file and commit before running validation!");
         }
         if (rc.failCount != 0) {
             throw new MojoExecutionException("Error formating '" + file + "' ");

